src/typegenPrinter.ts

```diff
@@ -523,6 +526,21 @@ export class TypegenPrinter {
     return returnTypeMap
   }
 
+  buildReturnTypeNamesMap() {
+    const returnTypeMap: TypeFieldMapping = {}
+    const hasFields: (GraphQLInterfaceType | GraphQLObjectType)[] = []
+    hasFields
+      .concat(this.groupedTypes.object)
```

Comment:
Only output types?

If so maybe that should be reflected in the name (see below for another name comment).

Reply:
I followed the existing naming scheme used for `NexusGenFieldTypes` interface, which doesn't precise either that it's only for interfaces & output objects. `NexusGenFieldTypes` -> `NexusGenFieldTypeNames`. Do you feel strongly we should rename them?

Reply:
No not now, sounds good.